### ISSUE TITLE
Executes constantly after upgrading from node 10 to 14

### ISSUE THREAD

**Issue by nouellette** [jaclarke/cronosjs#14]
The behavior changed between node 10 to 14.

10 works fine
14 will constantly execute with no delay

```
import {scheduleTask} from 'cronosjs'

scheduleTask('2 0 0 * * * *', () => {
    console.log(Date.now())
  }, {
    timezone: 'UTC',
    missingHour: 'offset',
  })
```

**Comment by ipatalas** [comment]
Oh, I've got quite similar issue.
See https://github.com/Steve-Mcl/node-red-contrib-cron-plus/issues/35 to see how it started.

Anyway, figured out it's related to `node.js` version.
Code to reproduce the issue:
```js
const CE = require('cronosjs').CronosExpression;
  
console.log(CE.parse('0 0 0 1 * ? *').nextDate().toLocaleString())
console.log(CE.parse('0 0 0 1 * ? *', {timezone: "Europe/Warsaw"}).nextDate().toLocaleString())
```

The output is as follows:
![image](https://user-images.githubusercontent.com/2623329/113344472-67422200-9331-11eb-9b3a-cdf2c9f737af.png)
I would expect them to be the same, both being the exact beginning of the month.

When I don't pass the timezone it works just fine. It works fine on node.js v10 or v12.
Try running that on `14.16.0` (not sure if other v14 cause the same problem).
You can change the timezone to yours so that the hours will match.

**Comment by jaclarke** [comment]
Sorry for the delay @nouellette, and @ipatalas thanks for tracking it down to the timezone handling.

I think I've found the root cause: between node.js v13.14.0 and v14.0.0, the `Intl.DateTimeFormat` api (which cronosjs uses to handle IANA timezones) has decided that midnight is now `24:00:00` instead of `00:00:00`, and that breaks everything else.
```js
console.log(process.version);

const dtf = new Intl.DateTimeFormat("en-US", {
  hour12: false,
  timeZone: "Europe/London",
  year: "numeric",
  month: "2-digit",
  day: "2-digit",
  hour: "2-digit",
  minute: "2-digit",
  second: "2-digit",
});

const midnight = new Date(2021, 03, 1);

console.log(midnight);

console.log(dtf.format(midnight));
console.log(dtf.formatToParts(midnight)[6]);
```
![image](https://user-images.githubusercontent.com/2085287/113354887-cdca3e80-9337-11eb-934e-877a66f380e8.png)

I'll publish a bugfix release shortly.

**Comment by jaclarke** [comment]
Fixed in v1.7.1
